### PR TITLE
IPFS and Filecoin page addition

### PR DESCRIPTION
**Pull request by har00ga**
This PR includes the initial reviewed draft for the IPFS and Filecoin page (ref. #72). Our next steps are to brainstorm use-case howtos for the bottom of the doc, followed by a final proofread.

### CODE REVIEW

File: docs/introduction/ipfs-and-filecoin.md [hunk 1]
```diff
@@ -11,17 +6,17 @@ Filecoin and IPFS are complementary protocols for storing and sharing data in th
...
+As mentioned: in IPFS, data is hosted by nodes pinning the desired data. To persist data when a user’s node is offline - users must either rely on other peers to voluntarily/altruistically pin their data, or  use a centralized pinning service (often running long lived host nodes on a  cloud-service provider) to store the data.  
+
+Relying on peers to altruistically cache data might work well where one -- or multiple -- organizations share popular files on an internal network (intranet). Or where strong social contracts can be used to ensure the content remains hosted and maintained long-term. Most users in the IPFS Network use a pinning service.
```
Comment (mishmosh) [inline docs/introduction/ipfs-and-filecoin.md:19]: 2nd sentence here is a fragment.

**Comment by mishmosh** [comment]
It looks like the title & description metadata got removed from the top of this page. IIUC they're important for rendering. @terichadbourne would know the specifics though.

File: docs/introduction/ipfs-and-filecoin.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+First, it is important to understand that Filecoin is built _on top of_  IPFS. Filecoin is intended to be a highly integrated and seamless storage market that _leverages_ the underlying functionality provided by IPFS; in this sense, they are connected, but can be implemented completely independent of each other. Users do not _need_ to interact with Filecoin in order to use IPFS.
+
+Some features within Filecoin that are shared with IPFS include:
+
+* The use of [IPLD](https://ipld.io/) for blockchain data structures
+* The use of [libp2p](https://libp2p.io/) by Filecoin nodes to establish secure connections with each other
+* Messaging between nodes and block propagation in Filecoin are facilitated by [libp2p](https://libp2p.io/) pubsub
+* CIDs in Filecoin and IPFS share a hashing specification
+
+ > particular use case sections and content will go here
```
[the remark this answers is not in the document]
Comment (har00ga) [reply docs/introduction/ipfs-and-filecoin.md]: Removed and note added!

**Comment by har00ga** [comment]
Ahhh, that was my bad! Forgot it was there and copy/pasted Molly's changes from earlier. Readding

**Pull request merged**